README.md

```diff
@@ -137,23 +152,26 @@ You can customize the automatic generated proxy with the following special label
 When all the values above are added to a service, the following configuration will be generated:
 ```
 service.example.com {
-	proxy /source https://servicename:8080/api
+  route /source/* {
+    uri strip_prefix /source
+    rewrite * /api{uri}
```

Comment:
> Hmm, I don't think this is exactly correct - rewrites apply to the path, not the uri.

I changed to /api{path} and it works well, code pushed.

> Also this specific example could be done with a uri replace instead of strip_prefix then rewrite. I think if both source and target are set, you can take that approach instead.

Wouldn't `uri replace` replace the text in any part of the uri/path?
I need to make sure it replaces on in the beginning.

> Also /source/* won't match requests to /source btw, so maybe you'd want to add a redir rule outside the route to do redir /source /source/ 308 ?

Good point, I was trying to to keep the same behavior as caddy v1 `proxy /source upstream`.
But I'm not sure if caddy v1 would also match /source or not. Do you know that by heart?

When necessary, users can do a rewrite by themselves like this example: https://github.com/lucaslorentz/caddy-docker-proxy/blob/caddy-v2/examples/example.yaml#L88

Reply:
Wait, I think I'm wrong, I think `{uri}` is right 🤦‍♂️ sorry (but I think both work ultimately)

https://github.com/caddyserver/caddy/blob/master/modules/caddyhttp/rewrite/rewrite.go#L207

Reply:
> Wait, I think I'm wrong, I think {uri} is right 🤦‍♂️ sorry (but I think both work ultimately)

{path} looks good. Indeed it supports both, as I don't want to change query, {path} is more appropriate.

Reply:
> Wouldn't uri replace replace the text in any part of the uri/path?
> I need to make sure it replaces on in the beginning.

Yep, replace operates on both the path and the raw query string.